confmap/confmap.go

```diff
@@ -603,6 +603,11 @@ func zeroSliceHookFunc() mapstructure.DecodeHookFuncValue {
 				to.Set(reflect.MakeSlice(to.Type(), from.Len(), from.Cap()))
 			}
 		}
+		if to.CanSet() && to.Kind() == reflect.Map && from.Kind() == reflect.Map {
```

Comment:
This is minor, but I wonder if we should change the name of the hook now that we're explicitly handling a type other than slices.